src/__tests__/JWT/CommonSignerTest/CommonSignerTest.ts

```diff
@@ -0,0 +1,205 @@
+import * as jwt from 'jsonwebtoken'
+import * as u8a from 'uint8arrays'
+import { Extensible, ServiceEndpoint } from 'did-resolver'
+
+// redeclare non-exported interface from did-resolver
+interface JsonWebKey extends Extensible {
+  alg?: string
+  crv?: string
+  e?: string
+  ext?: boolean
+  key_ops?: string[]
+  kid?: string
+  kty: string
+  n?: string
+  use?: string
+  x?: string
+  y?: string
+}
+
+interface privateJsonWebKey extends JsonWebKey {
+  d?: string
+}
+
+interface VerificationMethodLegacyCommonSigner {
+  id?: string
+  type: string
+  owner?: string
+  publicKey?: string
+  publicKeyBase58?: string
+  publicKeyBase64?: string
+  publicKeyJwk?: JsonWebKey
+  publicKeyHex?: string
+  publicKeyMultibase?: string
+  blockchainAccountId?: string
+  ethereumAddress?: string
+}
+
+interface VerificationMethodCommonSigner {
+  id: string
+  type: string
+  controller: string
+  publicKeyBase58?: string
+  publicKeyBase64?: string
+  publicKeyJwk?: JsonWebKey
+  publicKeyHex?: string
+  publicKeyMultibase?: string
+  blockchainAccountId?: string
+  ethereumAddress?: string
+}
+
+type DIDDocumentLegacyCommonSigner = {
+  '@context'?: 'https://www.w3.org/ns/did/v1' | string | string[]
+  id: string
+  alsoKnownAs?: string[]
+  controller?: string | string[]
+  verificationMethod?: VerificationMethodLegacyCommonSigner[]
+  service?: ServiceEndpoint[]
+  publicKey?: VerificationMethodLegacyCommonSigner[]
+  authentication?: VerificationMethodLegacyCommonSigner[]
+}
+
+type DIDDocumentCommonSigner = {
```

Comment:
Why is this necessary?

Reply:
Short answer. I found the did-resolver library limited in terms of its interfaces and wanted to package the creation of did documents in functions. It seemed like a bit of a hack. I will get back to this. Thank you.

Reply:
can you be more specific in where the interfaces are limiting?

The interfaces defined there are meant to mimic the DID spec, with some support for previous versions of the spec. If they are limiting, this might be to match the spec, but, if we missed something there, please raise an issue, instead of redefining these types.

Reply:
There were some other things in there like (https://github.com/bshambaugh/did-jwt/blob/master/src/__tests__/JWT/CommonSignerTest/CommonSignerTest.ts#L24-L75) that were added to avoid the use of any in the typescript file. This might need to be refactored. I guess I didn't realize how scary this was going to get.

Reply:
The use of `any` and `unknown` in the test files is very much acceptable, as well as type casting for the sake of making tests simple to understand, especially for testing corner cases, as long as there exist some tests that run through the types as they are expressed or expected by the API.

You shouldn't feel the need to redefine big chunks of API just for testing corner cases.

Since these libraries are ultimately boiled down to plain JavaScript, it is even better to have tests that don't fit the neatly defined types that the IDE can warn you about.

Reply:
I'm in the process of going through this, and thus made some changes to CommonSignerTest,ts, and could possibly do more. I posted a description of what I was up to in a slack channel, which helped guide my mind:
"CommonSignerTest.ts is an interesting file. It is a refactoring of part of https://github.com/decentralized-identity/did-jwt/blob/master/src/__tests__/JWT.test.ts . I split it out, while refactoring (the test, grin...).
Line 36 through 91 are interesting, and they were reformulated for use in a common file. Here is the new structure: https://github.com/bshambaugh/did-jwt/tree/master/src/__tests__/JWT . The const declarations that were
formally didDocLegacy, didDoc, and audDidDoc are now functions in ./CommonSignerTest/CommonSignerTest.ts in the new structure and CommonSignerTest.ts is called by JWT.ES256KSigner.test.ts and JWT.ES256Signer.test.ts .
I did this to avoid bloating of the test orignal test file. This makes my reviewer a bit bewildered. He is no longer sure what to trust. (https://github.com/decentralized-identity/did-jwt/pull/212#pullrequestreview-903264857).Along the way the typescript editor in the Visual Studio IDE started complaining about my use of any. Interfaces came to the rescue. I was not entirely satisfied with the interfaces that came with the did-resolver npm package.
(https://github.com/decentralized-identity/did-resolver/blob/master/src/resolver.ts) , so I created a few of my own (https://github.com/bshambaugh/did-jwt/blob/master/src/__tests__/JWT/CommonSignerTest/CommonSignerTest.ts#L20-L36)
for legacy verification method and the private key (reason given here: https://github.com/decentralized-identity/did-resolver/issues/121).
I also had to create some types for the legacy and more current did documents (https://github.com/bshambaugh/did-jwt/blob/master/src/__tests__/JWT/CommonSignerTest/CommonSignerTest.ts#L38-L62). In addition, I had to re-declare the
JsonWebKey interface since it was not exported from the did-resolver npm package (https://github.com/bshambaugh/did-jwt/blob/master/src/__tests__/JWT/CommonSignerTest/CommonSignerTest.ts#L6-L18)."

Reply:
tl;dr: maybe I should focus in on this:

"as well as type casting for the sake of making tests simple to understand, especially for testing corner cases, as long as there exist some tests that run through the types as they are expressed or expected by the API."

Do I need to write things in a different way? I had type any for CreatedidDocLegacy , CreatedidDoc and CreateauddidDoc , but went the interface route when I got complaints from my IDE.

Reply:
Okay, CommonSignerTest.ts  [1] is substantially simplified  [1] (https://github.com/decentralized-identity/did-jwt/pull/212/files#diff-0a517e08539420acec2253799dfa947d00c7d2e854614a743dd72a08b9fd5664)

Reply:
@mirceanis , CommonSignerTest.ts is substantially cleaned up. I reused declarations from did-resolver where possible.